gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/GitConfigListener.java

```diff
@@ -137,23 +118,23 @@ public void removeChange(DiffEntry change) {
           .withDescription(SPEC_DESCRIPTION)
           .build();
 
-        this.flowCatalog.remove(spec.getUri());
+      this.flowCatalog.remove(spec.getUri());
     }
   }
 
 
-    /**
-     * check whether the file has the proper naming and hierarchy
-     * @param configFilePath the relative path from the repo root
-     * @return false if the file does not conform
-     */
+  /**
+   * check whether the file has the proper naming and hierarchy
+   * @param configFilePath the relative path from the repo root
+   * @return false if the file does not conform
+   */
   private boolean checkConfigFilePath(String configFilePath) {
     // The config needs to stored at configDir/flowGroup/flowName.(pull|job|json|conf)
     Path configFile = new Path(configFilePath);
     String fileExtension = Files.getFileExtension(configFile.getName());
 
     if (configFile.depth() != CONFIG_FILE_DEPTH
-        || !configFile.getParent().getParent().getName().equals(folderName)
+        || !configFile.getParent().getParent().getName().equals(configBaseFolderName)
         || !(PullFileLoader.DEFAULT_JAVA_PROPS_PULL_FILE_EXTENSIONS.contains(fileExtension)
         || PullFileLoader.DEFAULT_JAVA_PROPS_PULL_FILE_EXTENSIONS.contains(fileExtension))) {
       log.warn("Changed file does not conform to directory structure and file name format, skipping: "
```

Comment:
I get not failing the entire service--but seems a particularly soft error to merely log a warning... how often do people notice this, I wonder...

Reply:
I agree the logging isn't very visible but its preferable to stopping the service, ideally we should add tests in the flowgraph itself / this git config repository so that improperly formatted folders never get read by GaaS.